### PR TITLE
auto transparent on scroll, zindex fix

### PR DESCRIPTION
**Pull request by Maxusha**

### CODE REVIEW

File: packages/header/src/header.tsx [hunk 1]
```diff
@@ -200,19 +202,37 @@ const HeaderBar = styled.div`
...
-  return (
-    <Header
-      {...props}
-      burgerClicked={clicked}
-      variant={isOpen ? 'dark' : props.variant}
-      isOpen={isOpen}
-    ></Header>
+
+  useScrollPosition(
+    (props: any) => {
```
Comment (raqystyle) [inline packages/header/src/header.tsx]: `any` ⛔️

Comment (Maxusha) [reply packages/header/src/header.tsx]: fixed in 625454a

File: packages/header/src/header.tsx [hunk 2]
```diff
@@ -8,15 +8,16 @@ import MobileMenu from './items/mobileMenu';
...
 
 export interface HeaderWrapperProps {
-  variant: 'transparent' | 'dark';
   imgPath: string;
   className?: string;
   phoneNumber: string;
 }
 
 export interface HeaderProps extends HeaderWrapperProps {
+  variant: 'transparent' | 'dark';
```
Comment (mrbabbs) [inline packages/header/src/header.tsx]: I think you can use `VariantType` here

File: packages/header/src/header.tsx [hunk 3]
```diff
@@ -200,19 +202,48 @@ const HeaderBar = styled.div`
...
-    <Header
-      {...props}
-      burgerClicked={clicked}
-      variant={isOpen ? 'dark' : props.variant}
-      isOpen={isOpen}
-    ></Header>
+
+  useScrollPosition(
+    ({ currPos }: ScrollProps) => {
+      const newVariant = currPos.y < 0 ? 'dark' : 'transparent';
```
Comment (mrbabbs) [inline packages/header/src/header.tsx]: I see we use in multiple place `dark` and `transparent` as string, I think is a good idea to create constants for those value and also export them

Comment (Maxusha) [reply packages/header/src/header.tsx]: fixed in 7f40081

File: packages/header/src/header.tsx (hunk 2 above)
Comment (Maxusha) [reply packages/header/src/header.tsx]: It's done using Enum in 7f40081

**Pull request merged**